test/feed.test.js

```diff
@@ -91,5 +124,62 @@ describe('Post data class tests', () => {
       await feed.setDelayed(60);
       expect(feed.isDelayed()).resolves.toBe(true);
     });
+
+    test('Updated feed should be different from data', async () => {
+      const feed = new Feed(data.author, data.url, data.user, data.link, null, null);
+      feed.author = 'Author Post';
+      feed.url = 'https://modified.user.feed.com/feed.rss';
+      await feed.update();
+      const modifiedFeed = await Feed.byId(feed.id);
+      // modifiedFeed.id should be different from feed.id as feed hasn't been reassigned to updated feed.id value
+      expect(modifiedFeed.id).not.toBe(feed.id);
+      expect(modifiedFeed.id).not.toBe(createFeed.id);
+      expect(modifiedFeed.url).not.toBe(createFeed.url);
+      expect(modifiedFeed.author).not.toBe(createFeed.author);
+      expect(modifiedFeed.url).toBe(feed.url);
+      expect(modifiedFeed.author).toBe(feed.author);
+      expect(modifiedFeed.id).toBe(urlToId(feed.url));
+    });
+
+    test('Removing feeds should also remove posts', async () => {
+      const feed = await Feed.byId(await Feed.create(data));
+      postData1.feed = feed;
+      postData2.feed = feed;
+      await Promise.all([
+        Post.create(postData1),
+        Post.create(postData2),
+        Elastic.indexPost(postData1.text, postData1.id),
+        Elastic.indexPost(postData2.text, postData2.id),
+      ]);
+
+      const posts = await Promise.all([Post.byId(postData1.id), Post.byId(postData2.id)]);
+      const elasticPosts = await Elastic.search();
+
+      // Make sure our feed is correct and saved
+      expect(feed.id).toBe(data.id);
+      // Make sure our posts actually have data and is the same as PostData1 + PostData2
+      expect(posts[0].id).toBe(postData1.id);
+      expect(posts[1].id).toBe(postData2.id);
+      expect(posts[0].feed.id).toBe(postData1.feed.id);
+      expect(posts[1].feed.id).toBe(postData2.feed.id);
+      expect(posts[0].feed.author).toBe(postData1.feed.author);
+      expect(posts[1].feed.author).toBe(postData1.feed.author);
+      // Testing Posts in ElasticSearch here
+      expect(elasticPosts.results).toBe(posts.length);
+      // Post doesn't contain text so we're checking for postData.text here instead
+      expect(Object.values(elasticPosts.values).indexOf(postData1.text)).not.toBe(-1);
+      expect(Object.values(elasticPosts.values).indexOf(postData2.text)).not.toBe(-1);
```

Comment:
If we do a better job in the results from es search() this can be simplified.

Reply:
Simplified!